kraftwerk-core/src/main/java/fr/insee/kraftwerk/core/outputs/scripts/ImportScript.java

```diff
@@ -1,37 +1,34 @@
-package fr.insee.kraftwerk.core.outputs;
-
-import java.util.LinkedHashMap;
-import java.util.Map;
-import java.util.Map.Entry;
+package fr.insee.kraftwerk.core.outputs.scripts;
 
 import fr.insee.kraftwerk.core.Constants;
 import fr.insee.kraftwerk.core.metadata.Variable;
 import fr.insee.kraftwerk.core.metadata.VariableType;
 import fr.insee.kraftwerk.core.metadata.VariablesMap;
-import fr.insee.vtl.model.Structured.DataStructure;
-import lombok.AllArgsConstructor;
-import lombok.Getter;
-
-/** POJO class to store information needed to write a script for a CSV table. */
-@AllArgsConstructor
-public class TableScriptInfo {
-
-	/** The name of the dataset in the destination language. */
-	@Getter
-	String tableName;
-	/** The CSV file name of the table. */
-	@Getter
-	String csvFileName;
-	/** The data structure (containing variable names and types) of the table. */
-	@Getter
-	DataStructure dataStructure;
-
-	/** The data structure (containing variable length) of the table. */
-	@Getter
-	Map<String, VariablesMap> metadataVariables;
-
-	public Map<String, Variable> getAllLength(DataStructure dataStructure,
-			Map<String, VariablesMap> metadataVariables) {
+import fr.insee.vtl.model.Structured;
+
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+
+/**
+ * Class to generate import scripts for the csv output tables.
+ * Some methods that could be implemented later: R base script and Python with pandas script.
+ */
+public abstract class ImportScript {
+
+	final List<TableScriptInfo> tableScriptInfoList;
+	static final String END_LINE = "\n";
+
+	/** @see TableScriptInfo */
+	protected ImportScript(List<TableScriptInfo> tableScriptInfoList) {
+		this.tableScriptInfoList = tableScriptInfoList;
+	}
+
+	public abstract String generateScript();
+
+	static Map<String, Variable> getAllLength(Structured.DataStructure dataStructure,
```

Comment:
(Sonar) 

> Refactor this method to reduce its Cognitive Complexity from 54 to the 15 allowed.

This method is very complex and not documented. What is does is a bit unclear, seems some things are hard-coded inside.